final_project/snapvisite/models.py

```diff
@@ -31,13 +32,18 @@ def __str__(self):
         return f'{self.category_name}'
 
 
+def save_photo(instance, filename):
+    filename = f'{uuid.uuid1()}_{filename}'
+    return f'company_photos/{instance.company_name}_{instance.id}/{filename}'
+
+
 class Company(models.Model):
     company_name = models.CharField(max_length=128)
-    photo = models.ImageField(blank=True, null=True, upload_to='company_photo/')
+    photo = models.ImageField(blank=True, null=True, upload_to=save_photo)
     description = models.TextField(blank=True, null=True)
     created_date = models.DateTimeField(auto_now_add=True)
     owner = models.ForeignKey(Profile, on_delete=models.DO_NOTHING)
-    address = models.OneToOneField(Address, on_delete=models.CASCADE)
+    address = models.OneToOneField(Address, on_delete=models.CASCADE, blank=True, null=True)
```

Comment:
blank dla `one_to_one` to chyba bez sensu, tego się raczej używa dla `charfield`, pytanie, czy może być firma bez adresu?